bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/client/Client.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.entsoe.internal.client;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.TreeMap;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.entsoe.internal.exception.entsoeConfigurationException;
+import org.openhab.binding.entsoe.internal.exception.entsoeResponseException;
+import org.openhab.binding.entsoe.internal.exception.entsoeUnexpectedException;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+/**
+ * @author Miika Jukka - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class Client {
+
+    private final Logger logger = LoggerFactory.getLogger(Client.class);
+
+    private DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+    // private @Nullable DocumentBuilder documentBuilder;
+
+    private Map<ZonedDateTime, Double> responseMap = new TreeMap<>();
+
+    public Map<ZonedDateTime, Double> doGetRequest(Request request, int timeout)
+            throws entsoeResponseException, entsoeUnexpectedException, entsoeConfigurationException {
+        try {
+            logger.debug("Sending GET request with parameters: {}", request);
+            String url = request.toUrl();
+            String responseText = HttpUtil.executeUrl("GET", url, timeout);
+            if (responseText == null) {
+                logger.error("GET request failed and returned null for request url: {}", url);
+                throw new entsoeResponseException("Request failed");
+            }
+            logger.debug("{}", responseText);
+            return parseXmlResponse(responseText);
+        } catch (IOException e) {
+            if (e.getMessage().contains("Authentication challenge without WWW-Authenticate header")) {
```

Comment:
You could consider using the common `httpClient` directly. This will give you more control, and you could avoid having to check this by a text comparison. See for example:
https://github.com/openhab/openhab-addons/blob/cae65393d41b4abae692d79c452883f06663ccbd/bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/IndegoController.java#L158-L171